tests/test_happy_path.py

```diff
@@ -0,0 +1,176 @@
+from brownie import Wei, chain, RewardsManager
+from utils.config import (
+    gift_index,
+    one_inch_token_owner,
+    farming_rewards_owner,
+    initial_rewards_duration_sec,
+    rewards_amount,
+)
+
+# Test Case:
+# 1. Deploy and configure rewards manager
+# 2. Reward farming contract sets reward manager as distributor for LDO token gift
+# 3. DAO transfers LDO to reward manager contract
+# 4. 1INCH distributor transfers tokens to farming contract directly
+# 5. Someone starts new reward period (1INCH via their distributor, LDO via reward manager)
+# 6. Users A and B are obtaining stETH and DAI tokens somehow
+# 7. Users A and B deposit their stETH and DAI tokens to liquidity pool to receive LP tokens
+# 8. Users A and B are staking their LP tokens to reward farming contract
+# 9. User A exists from reward farming contract after half of rewarding period
+# 10. User A withdraws tokens from liquidity pool
+# 11. User B exists from reward farming contract after full rewarding period
+# 12. User B withdraws tokens from liquidity pool
+# 13. User B has more rewards than A
+def test_happy_path(
+    accounts,
+    ape,
+    stranger,
+    ldo_token,
+    steth_token,
+    dai_token,
+    one_inch_token,
+    farming_rewards,
+    mooniswap,
+    dao_voting_impersonated,
+    dao_agent
+):
+    rewards_period = initial_rewards_duration_sec
+    dai_holder = "0xa405445ff6ed916b820a744621ef473b260b0c1c"
+
+    # Deploy and configure rewards manager
+    rewards_manager = RewardsManager.deploy(farming_rewards, {"from": ape})
```

Comment:
Its better to use deploy script here to test full deploy sequence

Reply:
Addressed in https://github.com/maddevsio/lido/pull/3/commits/45bda87b4f03a46c6ca9f252c8b6fa505f974e55